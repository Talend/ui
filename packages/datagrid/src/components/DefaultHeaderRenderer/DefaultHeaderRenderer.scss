$td-header-component-border-color: $gallery !default;
$td-quality-bar-hover-height: 1rem !default;

.td-header-component {
	border-right: $td-header-component-border-color 1px solid;
	height: 100%;

	&:hover :global(.td-quality-bar) {
		height: $td-quality-bar-hover-height;
	}

	.td-header {
		height: 100%;
		width: 100%;
		padding: 1rem;
		text-align: left;
		background-color: inherit;
		border: none;

		&-title {
			display: flex;
			color: $dove-gray;
			font-weight: 400;

			&-ellipse {
				flex-grow: 1;
				overflow: hidden;
				text-overflow: ellipsis;
				white-space: nowrap;
			}
		}

		&-type {
			font-size: 1.2rem;
			font-weight: 700;
			color: $dark-silver;
		}
	}
}
