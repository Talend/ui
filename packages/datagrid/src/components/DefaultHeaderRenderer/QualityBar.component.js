import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import theme from './QualityBar.scss';

const QUALITY_BAR_MIN_WIDTH = 5;

export default function QualityBar(props) {
	return (
		<div className={classNames(theme['td-quality-bar'], 'td-quality-bar')}>
			{!!props.invalid && (
				<span
					className={classNames(theme['td-quality-bar-invalid'], 'td-quality-bar-invalid')}
					style={{ width: `${props.invalid}%`, minWidth: `${QUALITY_BAR_MIN_WIDTH}%` }}
					title={`${props.invalid}%`}
				/>
			)}
			{!!props.empty && (
				<span
					className={classNames(theme['td-quality-bar-empty'], 'td-quality-bar-empty')}
					style={{ width: `${props.empty}%`, minWidth: `${QUALITY_BAR_MIN_WIDTH}%` }}
					title={`${props.empty}%`}
				/>
			)}
			{!!props.valid && (
				<span
					className={classNames(theme['td-quality-bar-valid'], 'td-quality-bar-valid')}
					style={{ width: `${props.valid}%`, minWidth: `${QUALITY_BAR_MIN_WIDTH}%` }}
					title={`${props.valid}%`}
				/>
			)}
		</div>
	);
}

QualityBar.propTypes = {
	valid: PropTypes.number,
	empty: PropTypes.number,
	invalid: PropTypes.number,
};
