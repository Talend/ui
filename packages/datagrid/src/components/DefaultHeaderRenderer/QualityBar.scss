$td-quality-bar-height: 0.5rem !default;

.td-quality-bar {
	display: flex;
	height: $td-quality-bar-height;
	width: 100%;
	padding: 0 0.3rem;
	position: absolute;
	bottom: 0;

	&-invalid {
		background-color: $td-quality-bar-invalid-background-color;
	}

	&-valid {
		background-color: $td-quality-bar-valid-background-color;
	}

	&-empty {
		background-color: $td-quality-bar-empty-background-color;
	}
}
