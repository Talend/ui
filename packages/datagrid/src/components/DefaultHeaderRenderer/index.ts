export { default } from './DefaultHeaderRenderer.component';
