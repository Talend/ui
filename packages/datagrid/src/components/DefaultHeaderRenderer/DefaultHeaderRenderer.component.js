import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import QualityBar, { QUALITY_PROPTYPE } from './QualityBar.component';
import theme from './DefaultHeaderRenderer.scss';
import {
	QUALITY_KEY,
	QUALITY_INVALID_KEY,
	QUALITY_EMPTY_KEY,
	QUALITY_VALID_KEY,
} from '../../constants';

export const HEADER_RENDERER_COMPONENT = 'headerGrid';

export default function DefaultHeaderRenderer({ column, displayName, onFocusedColumn, onKeyDown }) {
	const onHeaderClick = () => onFocusedColumn(column.colId);
	const onHeaderKeyDown = event => onKeyDown(event, column.colId);

	return (
		<div className={classNames(theme['td-header-component'], 'td-header-component')}>
			<button
				className={classNames(theme['td-header'], 'td-header')}
				onClick={onHeaderClick}
				onKeyDown={onHeaderKeyDown}
			>
				<div className={classNames(theme['td-header-title'], 'td-header-title')}>
					<span
						className={classNames(theme['td-header-title-ellipse'], 'td-header-title-ellipse')}
						title={displayName}
					>
						{displayName}
					</span>
					<span>...</span>
				</div>
				<div
					className={classNames(theme['td-header-type'], 'td-header-type')}
					title={column.colDef.type}
				>
					{column.colDef.type}
				</div>
			</button>
			{column.colDef[QUALITY_KEY] && (
				<QualityBar
					invalid={column.colDef[QUALITY_KEY][QUALITY_INVALID_KEY]}
					empty={column.colDef[QUALITY_KEY][QUALITY_EMPTY_KEY]}
					valid={column.colDef[QUALITY_KEY][QUALITY_VALID_KEY]}
				/>
			)}
		</div>
	);
}

DefaultHeaderRenderer.propTypes = {
	column: PropTypes.shape({
		colDef: PropTypes.shape({
			[QUALITY_KEY]: PropTypes.shape({
				[QUALITY_EMPTY_KEY]: PropTypes.shape(QUALITY_PROPTYPE),
				[QUALITY_INVALID_KEY]: PropTypes.shape(QUALITY_PROPTYPE),
				[QUALITY_VALID_KEY]: PropTypes.shape(QUALITY_PROPTYPE),
			}),
		}),
	}),
	displayName: PropTypes.string,
	onFocusedColumn: PropTypes.func,
	onKeyDown: PropTypes.func,
};
