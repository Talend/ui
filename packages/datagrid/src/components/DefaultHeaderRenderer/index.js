import DefaultHeaderRenderer, {
	HEADER_RENDERER_COMPONENT,
} from './DefaultHeaderRenderer.component';

export { HEADER_RENDERER_COMPONENT };
export default DefaultHeaderRenderer;
