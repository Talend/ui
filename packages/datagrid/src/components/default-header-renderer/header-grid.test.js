import React from 'react';
import { shallow } from 'enzyme';

import {
	TALEND_QUALITY_KEY,
	TALEND_QUALITY_EMPTY_KEY,
	TALEND_QUALITY_INVALID_KEY,
	TALEND_QUALITY_VALID_KEY,
} from '../constants';

import DefaultHeaderRenderer from './header-grid.component';

describe('#DefaultBooleanCellRenderer', () => {
	it('should render DefaultBooleanCellRenderer', () => {
		const onFocusedColumn = jest.fn();
		const wrapper = shallow(
			<DefaultHeaderRenderer
				onFocusedColumn={onFocusedColumn}
				column={{
					colId: 'colId',
					colDef: {
						type: 'string',
						[TALEND_QUALITY_KEY]: {
							[TALEND_QUALITY_INVALID_KEY]: 33,
							[TALEND_QUALITY_EMPTY_KEY]: 33,
							[TALEND_QUALITY_VALID_KEY]: 34,
						},
					},
				}}
				displayName="Title"
			/>,
		);
		wrapper.find('button').simulate('click');
		// then
		expect(wrapper.getElement()).toMatchSnapshot();
		expect(onFocusedColumn).toHaveBeenCalledWith('colId');
	});
});
