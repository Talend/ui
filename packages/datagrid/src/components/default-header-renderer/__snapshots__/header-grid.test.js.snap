// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#DefaultBooleanCellRenderer should render DefaultBooleanCellRenderer 1`] = `
<div
  className="theme-td-header-component td-header-component"
>
  <button
    className="theme-td-header td-header"
    onClick={[Function]}
  >
    <div
      className="theme-td-header-first-line td-header-first-line"
    >
      <span
        className="theme-td-header-column-label td-header-column-label"
        title="Title"
      >
        Title
      </span>
      <span>
        ...
      </span>
    </div>
    <div>
      string
    </div>
  </button>
  <QualityBar
    empty={33}
    invalid={33}
    valid={34}
  />
</div>
`;
